### PR TITLE
Implemented secure stores

### PR DESCRIPTION
**Pull request by annakocheshkova**
Makes a use of secure token stores like it's done in `appcenter-cli`:
- Windows - windows credential store
- MacOS - keychain
- Fallback - insecure file storage

Implements the upgrade from the file storage to secure storages for those users who accidently may have token files:
- When the token doesn't come from new cred store, trying to fetch it from file store and if successful, remove it from there.
- If removing token on logout fails, trying to clean the file store up.

### CODE REVIEW

File: src/data/tokenStore/osx/osx-token-store.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import * as rx from "rx-lite";
+import * as childProcess from "child_process";
+import * as es from "event-stream";
+import * as stream from "stream";
+
+import { TokenStore, TokenEntry, TokenKeyType, TokenValueType } from "../tokenStore";
+import { createOsxSecurityParsingStream, OsxSecurityParsingStream } from "./osx-keychain-parser";
+
+const securityPath = "/usr/bin/security";
+const serviceName = "appcenter-cli";
```
Comment (max-mironov) [inline src/data/tokenStore/osx/osx-token-store.ts]: why service name is "appcenter-cli"? 
Also the same is for line below.

File: src/data/tokenStore/index.ts [hunk 2]
```diff
@@ -25,5 +28,12 @@ const getTokenFilePath = (tokenFile: string) => {
...
-store = createFileTokenStore(getTokenFilePath(Constants.AppCenterTokenFileName));
+if (os.platform() === "win32") {
+  store = createWinTokenStore();
+} else if (os.platform() === "darwin") {
+  store = createOsxTokenStore();
+} else {
+  store = createFileTokenStore(getTokenFilePath(Constants.AppCenterTokenFileName));
+}
 export const tokenStore = store;
+export const fileTokenStore = createFileTokenStore(getTokenFilePath(Constants.AppCenterTokenFileName));
```
Comment (max-mironov) [inline src/data/tokenStore/index.ts]: why export again `fileTokenStore`?  On line 38 we export `tokenStore`, do we need additional export?

File: src/data/tokenStore/win32/win-token-store.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
+//
+// Implementation of token store that reads and writes to the Windows credential store.
+// Uses included "creds.exe" program to access the credential store.
+//
+
+import * as childProcess from "child_process";
+import { Observable, Observer } from "rx-lite";
+import * as stream from "stream";
+import * as es from "event-stream";
```
Comment (max-mironov) [inline src/data/tokenStore/win32/win-token-store.ts]: Do we need to add `event-stream` dependency to pakage.json?

File: src/data/tokenStore/index.ts (hunk 2 above)
Comment (annakocheshkova) [reply src/data/tokenStore/index.ts]: Yep for compatibility

File: src/data/tokenStore/osx/osx-token-store.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        }
+      });
+    });
+  }
+
+  public set(key: TokenKeyType, value: TokenValueType): Promise<void> {
+    const args = [
+      "add-generic-password",
+      "-a", key,
+      "-D", "appcenter cli password",
```
Comment (max-mironov) [inline src/data/tokenStore/osx/osx-token-store.ts]: Why `appcenter cli password`?

**Pull request merged**